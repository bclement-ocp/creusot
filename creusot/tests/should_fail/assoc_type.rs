extern crate creusot_contracts;
use creusot_contracts::*;

trait Tr {
    type A;
}

struct Assoc<T: Tr> {
    item: T::A,
}

fn uses<T: Tr>(x: Assoc<T>) {}
