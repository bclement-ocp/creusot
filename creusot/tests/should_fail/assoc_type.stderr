warning: unused import: `creusot_contracts::*`
 --> assoc_type.rs:2:5
  |
2 | use creusot_contracts::*;
  |     ^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused variable: `x`
  --> assoc_type.rs:12:16
   |
12 | fn uses<T: Tr>(x: Assoc<T>) {}
   |                ^ help: if this is intentional, prefix it with an underscore: `_x`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: struct `Assoc` is never constructed
 --> assoc_type.rs:8:8
  |
8 | struct Assoc<T: Tr> {
  |        ^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: function `uses` is never used
  --> assoc_type.rs:12:4
   |
12 | fn uses<T: Tr>(x: Assoc<T>) {}
   |    ^^^^

error[creusot]: associated types are unsupported in type declarations
 --> assoc_type.rs:9:5
  |
9 |     item: T::A,
  |     ^^^^^^^^^^

error: aborting due to previous error; 4 warnings emitted

